papermill/execute.py

```diff
@@ -41,24 +45,24 @@ def preprocess(self, nb, resources):
     # Reset the notebook.
     for cell in nb.cells:
```

Comment:
It seems like resetting the notebook in a papermill specific manner should be its isolated function. That way it's easier to unit test and this kind of metadata manipulation should definitely be unit tested.